node/_process/streams.mjs

```diff
@@ -101,9 +101,27 @@ export const stdout = stdio.stdout = createWritableStdioStream(
   "stdout",
 );
 
+function _adjustHighWaterMark() {
+  if (Deno.isatty?.(Deno.stdin?.rid)) return 0;
+
+  try {
+    // TODO(PolarETech): On Windows, isFile() returns true even for a stream,
```

Comment:
Can/should this be fixed lower down in the native code?

Reply:
Thanks for taking a look at this PR.
I am currently checking the native code as well, including the points you mentioned.